File: legacy/ChessBoard/Knight.cs
using UnityEngine;

namespace ChessSwap.ChessBoard
{
    /// <summary>
    /// Knight is a class used to refer to any Knight object meant to move between tiles of
    /// the chessboard.
    /// 
    /// <list type="Knight">
    /// <item>
    /// <term>Author</term>
    /// <description>Nathan Skalka (nathansmail@cox.net)</description>
    /// </item>
    /// <item>
    /// <term>Copyright</term>
    /// <description>Nathan Skalka (c) 2015</description>
    /// </item>
    /// </list>
    /// 
    /// </summary>
    public sealed class Knight : Chessman
    {
        /// <summary>
        /// This method returns all possible movement with origin at this script's
        /// gameObject.
        /// </summary>
        /// <returns>
        /// An array of potential movement Positions (2D).
        /// </returns>
        public override Index2D[] MoveablePositions()
        {
            return new Index2D[]
            {
                2*Index2D.up + Index2D.right,
                2*Index2D.up + Index2D.left,

                Index2D.up + 2*Index2D.right,
                Index2D.down + 2*Index2D.right,

                2*Index2D.down + Index2D.right,
                2*Index2D.down + Index2D.left,
                
                Index2D.up + 2*Index2D.left,
                Index2D.down + 2*Index2D.left
            };
        }

        /// <summary>
        /// This method returns the point value of matching three in a row.
        /// </summary>
        /// <returns>
        /// The point value of the Chessman object.
        /// </returns>
        public override int ScoreValue()
        {   return 40;  }
    }
}

File: src/javagames/game/chessboard/Knight.java
package javagames.game.chessboard;
import javagames.game.structs.Index2D;

public final class Knight extends Chessman {

    /**
     * @see Chessman#moveablePositions()
     * @return An array of potential movement Positions (2D).
     */
	@Override
    public Index2D[] moveablePositions() {
        return new Index2D[] {
            Index2D.add(Index2D.mul(2, Index2D.up), Index2D.right),
            Index2D.add(Index2D.mul(2, Index2D.up), Index2D.left),

            Index2D.add(Index2D.up, Index2D.mul(2, Index2D.right)),
            Index2D.add(Index2D.down, Index2D.mul(2, Index2D.right)),

            Index2D.add(Index2D.mul(2, Index2D.down), Index2D.right),
            Index2D.add(Index2D.mul(2, Index2D.down), Index2D.left),

            Index2D.add(Index2D.up, Index2D.mul(2, Index2D.left)),
            Index2D.add(Index2D.down, Index2D.mul(2, Index2D.left))
        };
    }

    /**
     * @see Chessman#scoreValue()
     * @return The point value of this Chessman object.
     */
    @Override
    public int scoreValue() {
        return 5;
    }
}
